discovery/consul/consul_test.go

```diff
@@ -88,3 +88,35 @@ func TestConsulCheckForChanges(t *testing.T) {
 		t.Errorf("%v should have changed after TTL expired.", id)
 	}
 }
+
+func TestConsulEnableTagOverride(t *testing.T) {
+	backend := "service-TestConsulEnableTagOverride"
+	consul, _ := NewConsulConfig("consul:8500")
+	service := &discovery.ServiceDefinition{
+		ID:        backend,
+		Name:      backend,
+		IPAddress: "192.168.1.1",
+		TTL:       1,
+		Port:      9000,
+		ConsulExtras: &discovery.ConsulExtras{
+			EnableTagOverride: true,
+		},
+	}
+	id := service.ID
+	if consul.CheckForUpstreamChanges(backend, "") {
+		t.Fatalf("First read of %s should show `false` for change", id)
+	}
+	consul.SendHeartbeat(service) // force registration
+	consul.SendHeartbeat(service) // write TTL
```

Comment:
You should be able to do without the second heartbeat now that https://github.com/joyent/containerpilot/pull/247 has been merged.